### PR TITLE
Add feature $manage_datadir

### PR DESCRIPTION
**Pull request by alexharv074**
See discussion in #553

### CODE REVIEW

File: manifests/instance.pp [hunk 1]
```diff
@@ -73,6 +78,7 @@
   $status             = $elasticsearch::status,
   $config             = undef,
   $configdir          = undef,
+  $manage_datadir     = true,
```
Comment (electrical) [inline manifests/instance.pp]: Might be easier to have this one default to $elasticsearch::manage_datadir ?
Then you also don't need the logic further down.

Comment (alexharv074) [inline manifests/instance.pp]: Are you sure?  If I have it default to $elasticsearch::manage_datadir, I will still have the problem that the user may override the default either in the elasticsearch class, or in the elasticsearch::instance type.  Of course, I'd rather just force to user to set it in the elasticsearch::instance type, however it appears to me from the design that the preference is to allow the user to set things in two places.

Comment (electrical) [inline manifests/instance.pp]: When using $elasticsearch::manage_datadir as a default value for $manage_datadir in the instance define it allows you to either set it to false for every instance you create, or you can override it per individual instance.
This way if the user has 2 or more instances, the value doesn't have to be duplicated for each instance.

Comment (alexharv074) [inline manifests/instance.pp]: Got it, I see what you're saying.  Let me fix that.

Comment (alexharv074) [inline manifests/instance.pp]: OK, updated.

File: manifests/instance.pp [hunk 2]
```diff
@@ -112,6 +118,14 @@
     $instance_configdir = $configdir
   }
 
+  # manage_dir setting
```
Comment (electrical) [inline manifests/instance.pp]: You can get rid of this logic now with the modification you made for the default value :-)

Comment (alexharv074) [inline manifests/instance.pp]: Alright, unless I'm going slightly 1.30am crazy.

The user may choose to set $manage_dir to false using:

```
elasticsearch::instance { 'es-01':
  manage_datadir => false,
}
```

In that case, $elasticsearch::manage_datadir will be true and $manage_datadir will be false, and we'll fall through to the 'else'.

Still not seeing how I can remove this, as much as I wish I'd not had to put it in in the first place.

Comment (alexharv074) [inline manifests/instance.pp]: Ah, sorry.  It is the 1.30am crazy.  Ok, I'll fix it in the morning. :)

Comment (electrical) [inline manifests/instance.pp]: Haha that's okay :-)

**Comment by electrical** [comment]
jenkins, test it

File: manifests/instance.pp (hunk 2 above)
Comment (alexharv074) [inline manifests/instance.pp]: Right, fixed.

**Comment by alexharv074** [comment]
Hi @electrical what is the status here.  Do you have any remaining concerns about this feature?

**Comment by electrical** [comment]
could you rebase the PR? it seems you got a commit of mine in your PR.

**Comment by alexharv074** [comment]
@electrical  Fyi I have refactored all the code here to use a config as recommended in the puppetlabs/lvm module.  After doing this, I have found that I no longer have the need for this feature.  Would you still like to merge this PR in anyway?

**Comment by alexharv074** [comment]
Anyhow I have rebased.

**Comment by alexharv074** [comment]
Actually, if combining with standard puppetlabs/lvm you would need:
```
$datadir = '/srv/es'
class { 'elasticsearch':
  datadir => $datadir,
}
Mount[$datadir] -> File[$datadir]
```

**Comment by electrical** [comment]
Hi,

If it all works without this modification i would rather not merge this in.
Feels it could cause confusion for people and might lead them in using other parts in a wrong way.
Thank you though for the contribution!

**Pull request closed without merging**